source/extensions/transport_sockets/alts/noop_transport_socket_callbacks.h

```diff
@@ -0,0 +1,34 @@
+#include "envoy/network/transport_socket.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace TransportSockets {
+namespace Alts {
+
+/**
+ * A TransportSocketCallbacks for wrapped TransportSocket object. Some
+ * TransportSocket implementation wraps another socket which does actual I/O.
+ * This class is used by the wrapped socket as its callbacks instead of the real
+ * connection to hold back callbacks from the underlying socket to connection.
+ */
+class NoOpTransportSocketCallbacks : public Network::TransportSocketCallbacks {
+public:
+  explicit NoOpTransportSocketCallbacks(Network::TransportSocket* parent) : parent_(parent) {}
```

Comment:
https://github.com/envoyproxy/envoy/pull/4172#discussion_r210638046

Yeah if you delay creating TransportSocketCallbacks, i.e. in parent setTransportSocketCallbacks, you should be able to just take `TransportSocketCallbacks&` here.

Reply:
That's a good point! Done. 
I still keep the callbacks() in TransportSocket interface, as it's convenient to access to callbacks_ for underlying socket.